lib/src/main/kotlin/com/swmansion/starknet/data/types/HttpBatchRequestType.kt

```diff
@@ -0,0 +1,6 @@
+package com.swmansion.starknet.data.types
+
+import kotlinx.serialization.Serializable
+
+@Serializable
+sealed interface HttpBatchRequestType
```

Comment:
> Why is this called HttpBatchRequestType?

`HttpBatchRequestType` is the common return type of `batchRequestsOfDifferentTypes`. It's name can be improved, do you have any suggestions?

> From what I understand, the same interface is used for singular requests as well?

No, in case of singular requests and batch requests of the same type we want to the types to be fully inferred. Therefore it's not used there.

Reply:
Why `HttpRequestType` though? Afaik all the classes that implement this interface are **Responses**, not **Requests** 😄

I would go for something like:
1. `Response`
2. `StarknetResponse`
3. `ResponseBase`

Reply:
Reasonable, let's choose `StarknetResponse` among all proposals - it's more appropriate 😄

Reply:
Let's use StarknetResponse, as Response is a part of stdlib afaik